test/functional/search_controller_test.rb

```diff
@@ -317,6 +317,19 @@ def no_results
     assert_equal "1",       @response.headers["X-Slimmer-Result-Count"]
   end
 
+  test "should pass the search variant through when one is present" do
+    Services.rummager.expects(:search).with(has_entry(:match_length_variant, "B")).returns(response([]))
+    SearchAPI.unstub(:new)
+    @request.headers["HTTP_GOVUK_ABTEST_SEARCHMATCHLENGTH"] = "B"
```

Comment:
You can use the `with_variant` test helper instead of adding this header directly. It also confirms that the Vary header and meta tag have been added correctly.